spec/unit/puppet/provider/mysql_user/mysql_spec.rb

```diff
@@ -103,12 +148,19 @@
   end
 
   describe 'password_hash=' do
-    it 'changes the hash' do
+    it 'changes the hash < 5.7.6' do
       provider.expects(:mysql).with([defaults_file, '-e', "SET PASSWORD FOR 'joe'@'localhost' = '*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5'"]).returns('0')
 
       provider.expects(:password_hash).returns('*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5')
       provider.password_hash=('*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5')
     end
+    it 'changes the hash >= 5.7.6' do
+      provider.class.instance_variable_set(:@mysqld_version_string, '5.7.6')
+      provider.expects(:mysql).with([defaults_file, '-e', "ALTER USER 'joe'@'localhost' IDENTIFIED WITH mysql_native_password AS '*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5'"]).returns('0')
+
+      provider.expects(:password_hash).returns('*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5')
+      provider.password_hash=('*6C8989366EAF75BB670AD8EA7A7FC1176A95CEF5')
```

Comment:
When you're already in there, at the top of the tests, there is a `newhash` defined which has this value, I think this should use that then?